docs/guides/random-data-for-security.md

```diff
@@ -0,0 +1,128 @@
+# Using Random Data for Security
+
+Any secret token that allows someone to access a resource or perform an action should be generated with a secure
+random number generator.  That includes:
+
+- encryption keys
+- message padding and initialisation vectors
+- password generation
+- password salts and peppers
+- authentication tokens
+- session tokens
+- nonces
+
+The `[rand](https://perldoc.perl.org/functions/rand)` function is not fit for security purposes: it is seeded by only
+32-bits (4 bytes), and the [output can be predicted easily](https://www.perlmonks.org/?node_id=151595).
+
+A cryptographic-strength pseudo-random number generator (PRNG) won't improve security if it was seeded with data from
+`rand`: ultimately the output is of that algorithm still comes from a 32-bit seed.
+
+## Good Sources of Random Data
+
+Modern operating systems provide access to random data:
+
+- Linux and BSD variants have provided the `/dev/random` and `/dev/urandom` devices.
+- Newer Linux and BSD variants have the [getrandom(2)](http://man.he.net/man2/getrandom) system call.
+- Windows provides a `CryptGenRandom` function in the API.
+
+These sources are easy to access from Perl using several modules.  We are listing a few here that are lightweight,
+portable and (generally) have good defaults.
+
+### Crypt::URandom
+
+The simplest to use, and most portable, is [Crypt::URandom](https://metacpan.org/pod/Crypt::URandom).  It is a
```

Comment:
This may no longer be a pure-Perl version in the next release https://github.com/david-dick/crypt-urandom

Reply:
Might as well remove the "most portable" claim now, then :-|